File: Algorithms/Implementation/Sock Merchant/x.erl
% delete list element
% return deleted num
deleteListElem([],Elem) -> 0;
deleteListElem([Elem|T],Elem) ->
    deleteListElem(T) + 1;
deleteListElem([H|T],Elem) ->
    deleteListElem(T).

% Complete the sockMerchant function below.
% [1,2,1,2,1,3,2]
sockMerchant(_, []) -> [];
sockMerchant(_, [H|T]) ->

File: Algorithms/Implementation/Sock Merchant/x.lua
-- https://www.hackerrank.com/challenges/sock-merchant/problem

-- Complete the sockMerchant function below.
function sockMerchant(n, ar)
    local m = {}
    local num = 0
    for _, v in ipairs(ar) do
        if m[v] then
            m[v] = false
            num = num + 1
        else
            m[v] = true
        end
    end
    return num
end